### PR TITLE
AO-21098-Add-Support-for-Node-18-Replacing-17

### PR DESCRIPTION
**Pull request by ronilan**
## Overview

This pull request adds support or Node 18 replacing Node 17.

## Changes

Changes are limited to dev setup and GitHub Actions workflows.

## Issues

- The `restify` probe tests fail with node 18. The probelm is with `restify` itself, **not** with the probe. See: https://github.com/restify/node-restify/issues/1888. Tests disabled for Node 18.
- Results of Document workflow showed support for `restify` versions `4.1.0-6.2.0` with Node 10, 12, 14, 16, and 17. However, the workflow itself has a known issue with `restify` (see: AO-20494). **Versions 7.x and 8.x are supported by Node 16 and were supported by 17**. The error with Node 18 is a new issue with the package.

## Tests

All [acceptance tests pass](https://github.com/appoptics/appoptics-apm-node/actions/runs/2193042961).
Document workflow [ran as expected](https://github.com/appoptics/appoptics-apm-node/actions/runs/2203008424). [Supported components](https://github.com/appoptics/appoptics-apm-node/blob/test-each-version-2022-04-21-09-58-06/docs/supported-components.human) data generated.

### CODE REVIEW

**Review by cheempz — COMMENTED** [review]
LGTM except one comment @ronilan :)

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -35,10 +35,10 @@ jobs:
       - name: Checkout ${{ github.ref }}
         uses: actions/checkout@v2
 
-      - name: Setup Node ${{ github.event.inputs.node || '14' }}
+      - name: Setup Node ${{ github.event.inputs.node || '16' }}
```
[the remark this answers is not in the document]
Comment (ronilan) [reply .github/workflows/push.yml:38]: Thanks! Fixed with e7552cb25090cdd4a8d844fae9b334bb9e3f9542

**Review by cheempz — DISMISSED** [review]
LGTM, thanks @ronilan!

**Review by cheempz — APPROVED** [review]
LGTM, thanks @ronilan!  Curious, is the additional check in `test/versions.js` required in addition to the semver check in `test/probes/restify.tests.js` or an optimization?

**Pull request merged**